client/state/sites/selectors.js

```diff
@@ -281,6 +282,24 @@ export function isSitePreviewable( state, siteId ) {
 }
 
 /**
+ * Returns true if the site can be upgraded by the user, false if the
+ * site cannot be upgraded, or null if upgrade ability cannot be
+ * determined.
+ *
+ * @param  {Object}   state  Global state tree
+ * @param  {Number}   siteId Site ID
+ * @return {?Boolean}        Whether site is upgradeable
+ */
+export function isSiteUpgradeable( state, siteId ) {
+	// Cannot determine site upgradeability if there is no current user
+	if ( ! state.currentUser || ! getRawSite( state, siteId ) ) {
```

Comment:
I think we should not be directly accessing another state subtree here - how about using one of the `getCurrentUser` / `getCurrentUserId` selectors instead?